### PR TITLE
Add extra documentation for running compiler tests on Windows

### PR DESCRIPTION
**Pull request by Mordax**
### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [ ] New tests have been created which fail without the change (if possible) //Doc change, not code change
- [x] README / documentation was extended, if necessary
- [ ? ] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### Description

Hi there, new to the project and quite excited about Solidity. As a stubborn (and lacking other environment options) Windows user, I found that there was an important difference in the statement used for running Solidity tests, so I've added a note for Windows users on where they have to look to run the tests. Hopefully it'll save some time for some developers. 

I've done my best to follow the practices laid out in the contributing docs, and I'm interested in further contributing documentation and other fixes. Let me know if I've missed something or forgot a protocol. I didn't update the changelog as I didn't really notice documentation being common, but I'm more than happy to add it in if need be.

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -82,6 +82,12 @@ To run a basic set of tests that neither require ``aleth`` nor ``libz3``, run
 ``./scripts/soltest.sh --no-ipc --no-smt``. This script will run ``build/test/soltest``
 internally.
 
+.. note ::
+
+    Those working in a Windows environment wanting to run the above basic sets without aleth or libz3 in Git Bash, you would have to do
+    a different command: ``./build/test/RelWithDebInfo/soltest.exe -- --no-ipc --no-smt``. Make sure you followed the `building from source <https://solidity.readthedocs.io/en/latest/installing-solidity.html#building-from-source>`_
+    and that your Cmake was successful. If you're running this in plain Windows Command line, use ``.\build\test\RelWithDebInfo\soltest.exe -- --no-ipc --no-smt``.
```
Comment (christianparpart) [inline docs/contributing.rst]: `RelWithDebInfo` inclines that users did build in that mode on Windows, however, all others are supported too.

File: docs/contributing.rst [hunk 2]
```diff
@@ -82,6 +82,12 @@ To run a basic set of tests that neither require ``aleth`` nor ``libz3``, run
 ``./scripts/soltest.sh --no-ipc --no-smt``. This script will run ``build/test/soltest``
 internally.
 
+.. note ::
+
+    Those working in a Windows environment wanting to run the above basic sets without aleth or libz3 in Git Bash, you would have to do
```
Comment (chriseth) [inline docs/contributing.rst]: Thanks for the addition! Could you compress the paragraph a little? Also I think that people should know themselves that you have to build the tests first and can translate the `/` and `\` path notations - shouldn't `/` work on windows in general nowadays?

File: docs/contributing.rst (hunk 1 above)
Comment (christianparpart) [reply docs/contributing.rst]: This also applies to non-Windows development platforms. `soltest.sh` MAY also become superfluous since at least the `--testpath` logic was just lately merged into the soltest program itself, not requiring `soltest.sh` to do that magic anymore.

What *is* different on Windows however is how to access the aleth's IPC socket.

**Review by christianparpart — CHANGES_REQUESTED** [review]

File: docs/contributing.rst (hunk 2 above)
Comment (christianparpart) [reply docs/contributing.rst]: @chriseth to answer your windows-path-seperator question, the Windows Command Prompt only knows back-slash (`\`) but the PowerShell accepts both (`/` and `\`).

File: docs/contributing.rst (hunk 1 above)
Comment (chriseth) [reply docs/contributing.rst]: soltest is still more convenient, since you do not have to care about splitting into boost and non-boost options and can use the `--debug` switch.

File: docs/contributing.rst (hunk 2 above)
Comment (Mordax) [reply docs/contributing.rst]: @chriseth Quick clarification - by compressing, do you want me to make this less wordy or remove spacing? I can also eliminate the command prompt line and just state to be mindful of slashes, because as @christianparpart said, depending on where you run them, certain slash directions don't work.

File: docs/contributing.rst (hunk 1 above)
Comment (Mordax) [reply docs/contributing.rst]: @chriseth @christianparpart Should I keep the run test command or should I change up some flags?

File: docs/contributing.rst (hunk 2 above)
Comment (chriseth) [reply docs/contributing.rst]: just less wordy and probably good to keep the command - when I read such documentation, I mostly just look at the commands and if I want to know more, I read the text in addition to that

**Comment by ChrisChinchilla** [comment]
Thanks for the contribution @Mordax I am also very keen to make the onboarding process easier for everyone. I'm going to test this all myself on a Windows VM and will let you know if I have any suggestions.

**Review by chriseth — DISMISSED** [review]
Great!

**Comment by Mordax** [comment]
@ChrisChinchilla Awesome, let me know if you run into anything and I can confirm on my end.

File: docs/contributing.rst [hunk 3]
```diff
@@ -82,6 +82,11 @@ To run a basic set of tests that neither require ``aleth`` nor ``libz3``, run
 ``./scripts/soltest.sh --no-ipc --no-smt``. This script will run ``build/test/soltest``
 internally.
 
+.. note ::
+
+    Those working in a Windows environment wanting to run the above basic sets without aleth or libz3 in Git Bash, you would have to do: ``./build/test/RelWithDebInfo/soltest.exe -- --no-ipc --no-smt``.
```
Comment (axic) [inline docs/contributing.rst:87]: Wouldn't it be better to rephrase this to show that `soltest` can be used directly without the script?

**Comment by ChrisChinchilla** [comment]
@Mordax @chriseth I found some confusion with the various MS dependencies needed but couldn't quite pinpoint what I had forgotten to install. I may test again in the future. If you err on the side of install more than you need everything builds correctly.

@Mordax One question for you. I then looked at how to run tests under Windows. Have you had any success with those, and would you be willing to submit a PR maybe clarifying how Windows users can do that?

**Comment by Mordax** [comment]
@ChrisChinchilla Oh, I'm sorry, I didnt realize you pinged me so long ago! Maybe it's too long ago but I recall having some issues with running tests on Windows and then magically getting it to work. What problem did you run into? It may jog my memory and if it does, I'll do another PR.

**Comment by ChrisChinchilla** [comment]
@Mordax Mostly just knowing how to get them started I think.
